asmcomp/asmgen.ml

```diff
@@ -111,7 +111,8 @@ let rec regalloc ~ppf_dump round fd =
                 ": function too complex, cannot complete register allocation");
   dump_if ppf_dump dump_live "Liveness analysis" fd;
   let num_stack_slots =
-    if !use_linscan then begin
+    if !use_linscan ||
+       List.mem Cmm.Use_linscan_regalloc fd.Mach.fun_codegen_options then begin
```

Comment:
Another option would be to handle `!Clflags.use_linscan` completely in cmmgen.ml (by emitting this new `Use_linscan_regalloc` codegen option for all functions when `use_linscan` is set), which makes the code simpler here. Is there a reason for preferring the current approach?